tests/test_functions.py

```diff
@@ -110,3 +112,71 @@ def test_extract_hashbits_info():
             os.remove(fn)
         except OSError as e:
             print >>sys.stderr, '...failed to remove {fn}'.format(fn)
+
+
+FakeFQRead = collections.namedtuple('Read', ['name', 'accuracy', 'sequence'])
+FakeFastaRead = collections.namedtuple('Read', ['name', 'sequence'])
+
+
+def test_check_is_pair_1():
+    read1 = FakeFQRead(name='seq', accuracy='###', sequence='AAA')
+    read2 = FakeFQRead(name='seq2', accuracy='###', sequence='AAA')
+
+    assert not check_is_pair(read1, read2)
+
+
+def test_check_is_pair_2():
+    read1 = FakeFQRead(name='seq/1', accuracy='###', sequence='AAA')
+    read2 = FakeFQRead(name='seq/2', accuracy='###', sequence='AAA')
+
+    assert check_is_pair(read1, read2)
+
+
+def test_check_is_pair_3():
+    read1 = FakeFQRead(name='seq 1::', accuracy='###', sequence='AAA')
+    read2 = FakeFQRead(name='seq 2::', accuracy='###', sequence='AAA')
+
+    assert check_is_pair(read1, read2)
+
+
+def test_check_is_pair_4():
```

Comment:
Since we are using nose for tests, what about using the [raises][1] decorator?

``` python
from nose.tools import raises

@raises(ValueError)
def test_check_is_pair_4():
    read1 = FakeFQRead(name='seq/1', accuracy='###', sequence='AAA')
    read2 = FakeFastaRead(name='seq/2', sequence='AAA')

    check_is_pair(read1, read2)
```

Again using nose, this time the [assert_raises][2] context manager:

``` python
from nose.tools import assert_raises

def test_check_is_pair_4():
    read1 = FakeFQRead(name='seq/1', accuracy='###', sequence='AAA')
    read2 = FakeFastaRead(name='seq/2', sequence='AAA')

    with assert_raises(ValueError):
        check_is_pair(read1, read2)
```

Both seem more readable than using the ```try/assert False``` approach.

[1]: http://nose.readthedocs.org/en/latest/testing_tools.html#nose.tools.raises
[2]: http://stackoverflow.com/a/19310416